cli/mender-artifact/partition.go

```diff
@@ -130,6 +132,13 @@ func (v vImage) Open(comp artifact.Compressor, key []byte, imgname string) (VPIm
 			return nil, errors.Wrap(err, "can not process artifact")
 		}
 
+		var comp artifact.Compressor
+		if len(overrideCompressor) == 1 {
+			comp = overrideCompressor[0]
```

Comment:
Don't really think it's necessary to be too defensive here, we control everywhere this is called.